pkgs/kswitch/kswitch.sh

```diff
@@ -181,6 +217,42 @@ get_aws_credentials() {
         jq -r '.data | "export AWS_ACCESS_KEY_ID=\(.access_key); export AWS_SECRET_ACCESS_KEY=\(.secret_key); export AWS_SESSION_TOKEN=\(.security_token)"' > "${awsCredsFilePath}"
 }
 
+check_cache_timeout(){
+    # Check whether CACHE_TIMEOUT is an integer
+    [ "${CACHE_TIMEOUT}" -eq "${CACHE_TIMEOUT}" ] 2>/dev/null && return
+    log-error "CACHE_TIMEOUT has to be an integer !" && exit 1
+}
+
+get_credentials_or_cache() {
+    local toRefresh
+    toRefresh=${forceRefresh}
+    while true; do
+        if [ -f "${cachedExecCredentialsPath}" ] && [ "${toRefresh}" -eq 0 ]; then
+            local cachedResponse
+            cachedResponse="$(cat "${cachedExecCredentialsPath}")"
+            local cacheTimeout
+            cacheTimeout="$(jq -r '.cacheTimeout' <<<"${cachedResponse}")"
+            local cacheTimeoutEpoch
+            cacheTimeoutEpoch="$(date --date="${cacheTimeout}" +"%s")"
+            local nowEpoch
+            nowEpoch="$(date +"%s")"
+            if [ "${nowEpoch}"  -lt "${cacheTimeoutEpoch}" ]; then
+                log_debug "Fetching credentials from cache"
+                jq -r '.execCredential' <<<"${cachedResponse}"
+                return
+            fi
+        fi
+        local lockName
+        lockName="credentials_${zone}_lock"
+        lock "${lockName}" || continue
```

Comment:
What's the idea behind locks here ?

Reply:
I'm limiting calls to `get_credentials()`, and locking to prevent concurrent writing on `${cachedExecCredentialsPath}.tmp`